lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -15,6 +15,35 @@ defmodule Mix.Tasks.Compile.Elixir do
   Note it is important to recompile a file's dependencies as
   there are often compile time dependencies between them.
 
+  ## `__mix_recompile__?/0`
+
+  A module may export a `__mix_recompile__?/0` function that can
+  cause the module to be recompiled using custom rules. For example,
+  `@external_resource` already adds a compile-time dependency on an
+  external file, however to depend on a _dynamic_ list of files we
+  can do:
+
+      defmodule MyModule do
+        paths = Path.wildcard("*.txt")
+        paths_hash = :erlang.md5(paths)
+
+        for path <- paths do
+          @external_resource path
+        end
+
+        def __mix_recompile__?() do
+          Path.wildcard("*.txt") |> :erlang.md5() == unquote(paths_hash)
```

Comment:
Should this be negated?

```suggestion
          Path.wildcard("*.txt") |> :erlang.md5() != unquote(paths_hash)
```